### PR TITLE
fix: theme editor edit error

### LINKED ISSUE #42448: Theme Editor syntax error on edit theme config
### Reproduction link

[https://ant.design/theme-editor](https://ant.design/theme-editor)

### Steps to reproduce

Paste the following configuration in the "edit Theme Config" editor:
```
{
  "token": {
    "colorPrimary": "#1890ff",
    "colorSuccess": "#b7eb8f",
    "colorWarning": "#ffe58f",
    "colorError": "#f5222d",
    "colorInfo": "#9cdcfe",
    "colorTextBase": "#666666",
    "colorBgBase": "#ffffff",
    "colorBgLayout": "#f0f0f0",
    "fontSize": 14,
    "margin": 16,
    "padding": 16,
    "borderRadius": 4,
    "colorBgElevated": "#ffffff",
    "wireframe": false
  }
}
```

### What is expected?

The theme editor applies the theme configuration correctly.

### What is actually happening?

The theme editor returns "The theme of the JSON format is incorrect".

| Environment | Info |
| --- | --- |
| antd | 5.5.0 |
| React | - |
| System | Ubuntu 22.04.2 |
| Browser | Google Chrome 113.0.5672.92 (Official Build) (64-bit) |

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix https://github.com/ant-design/ant-design/issues/42448

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix: theme editor edit error        |
| 🇨🇳 Chinese |     修复主题编辑器错误      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 4e63956</samp>

Improved the theme editor functionality and stability by fixing error handling, optimizing rendering, and locking `vanilla-jsoneditor` dependency.

### <samp>🤖 Generated by Copilot at 4e63956</samp>

*  Simplify and fix content format error checking for theme editor ([link](https://github.com/ant-design/ant-design/pull/42452/files?diff=unified&w=0#diff-ef29a310ce5cad95573716c19d940bb06da30b1a8e571b8d7f6154dd8c42d92eL99-R109))
*  Render JSONEditor component only when edit modal is open to improve performance ([link](https://github.com/ant-design/ant-design/pull/42452/files?diff=unified&w=0#diff-ef29a310ce5cad95573716c19d940bb06da30b1a8e571b8d7f6154dd8c42d92eL170-R175))
*  Sort imports alphabetically in `.dumi/pages/theme-editor/index.tsx` for readability ([link](https://github.com/ant-design/ant-design/pull/42452/files?diff=unified&w=0#diff-ef29a310ce5cad95573716c19d940bb06da30b1a8e571b8d7f6154dd8c42d92eL1-R6))
*  Lock vanilla-jsoneditor dependency to version 1.4.0 in `package.json` to avoid breaking changes ([link](https://github.com/ant-design/ant-design/pull/42452/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L298-R298))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -295,7 +295,7 @@
     "ts-node": "^10.8.2",
     "typedoc": "^0.24.6",
     "typescript": "~5.0.2",
-    "vanilla-jsoneditor": "^0.17.1",
+    "vanilla-jsoneditor": "~0.17.1",
```
Comment (afc163) [inline package.json:298]:
```suggestion
    "vanilla-jsoneditor": "^0.17.1",
```

这个不用改。

File: .dumi/pages/theme-editor/index.tsx [hunk 2]
```diff
@@ -96,18 +96,17 @@ const CustomTheme = () => {
...
-      Array.isArray(status.contentErrors.validationErrors) &&
-      status.contentErrors.validationErrors.length === 0
-    ) {
-      setEditThemeFormatRight(true);
-    } else {
+    const contentFormatError = status.contentErrors;
+    if (contentFormatError) {
       setEditThemeFormatRight(false);
+    } else {
+      setEditThemeFormatRight(true);
```
Comment (MadCcc) [inline .dumi/pages/theme-editor/index.tsx:103]:
```suggestion
      setEditThemeFormatRight(!status.contentErrors);
```
这样可以不

File: .dumi/pages/theme-editor/index.tsx [hunk 3]
```diff
@@ -95,19 +98,18 @@ const CustomTheme = () => {
...
-      Array.isArray(status.contentErrors.validationErrors) &&
-      status.contentErrors.validationErrors.length === 0
-    ) {
-      setEditThemeFormatRight(true);
-    } else {
+    if (status.contentErrors) {
       setEditThemeFormatRight(false);
+    } else {
+      setEditThemeFormatRight(true);
+      setThemeConfigContent(newcontent);
```
Comment (MadCcc) [inline .dumi/pages/theme-editor/index.tsx:105]: 这里放到条件语句中会数据不一致吧，有错误就不更新 content 应该不是预期行为

File: .dumi/theme/common/JSONEditor/index.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ const Editor: React.FC<JSONEditorPropsOptional> = (props) => {
 
   useEffect(() => {
     editorRef.current?.updateProps(props);
```
Comment (yoyo837) [inline .dumi/theme/common/JSONEditor/index.tsx:20]: 只与 props.content 相关？要不 updateProps 直接改为接收 content？

Comment (BoyYangzai) [reply .dumi/theme/common/JSONEditor/index.tsx:20]: 这个我试了一下 updateProps 只更新content会导致他失去onchange的prop

File: .dumi/pages/theme-editor/index.tsx [hunk 5]
```diff
@@ -64,21 +64,19 @@ const CustomTheme = () => {
     json: undefined,
   });
 
-  useLayoutEffect(() => {
+  useEffect(() => {
     const storedConfig = localStorage.getItem(ANT_DESIGN_V5_THEME_EDITOR_THEME);
     if (storedConfig) {
-      setTheme(() => JSON.parse(storedConfig));
+      const tehemConfig = JSON.parse(storedConfig);
```
Comment (MadCcc) [inline .dumi/pages/theme-editor/index.tsx:70]:
```suggestion
      const themeConfig = JSON.parse(storedConfig);
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**